dev/tasks/tests/dev-init.js

```diff
@@ -0,0 +1,119 @@
+			expect( spies.npmInstall.called ).to.equal( false );
+			expect( spies.installGitHooks.called ).to.equal( false );
+		} );
+
+		it( 'clones repositories if no directories are found', () => {
+			const packageJSON = {
+				dependencies: {
+					'ckeditor5-core': 'ckeditor/ckeditor5-core',
+					'ckeditor5-plugin-devtest': 'ckeditor/ckeditor5-plugin-devtest',
+					'non-ckeditor-plugin': 'other/plugin'
+				}
+			};
+
+			initTask( mainRepositoryPath, packageJSON, options, emptyFn, emptyFn );
+			expect( spies.getDependencies.calledOnce ).to.equal( true );
+			expect( spies.getDependencies.firstCall.args[ 0 ] ).to.equal( packageJSON.dependencies );
+			expect( spies.getDirectories.calledOnce ).to.equal( true );
+			expect( spies.getDirectories.firstCall.args[ 0 ] ).to.equal( path.join( mainRepositoryPath, options.workspaceRoot ) );
+			expect( spies.parseRepositoryUrl.calledTwice ).to.equal( true );
+			expect( spies.cloneRepository.calledTwice ).to.equal( true );
+			expect( spies.cloneRepository.firstCall.args[ 0 ] ).to.equal( spies.parseRepositoryUrl.firstCall.returnValue );
+			expect( spies.cloneRepository.firstCall.args[ 1 ] ).to.equal( workspacePath );
+			expect( spies.cloneRepository.secondCall.args[ 0 ] ).to.equal( spies.parseRepositoryUrl.secondCall.returnValue );
+			expect( spies.cloneRepository.secondCall.args[ 1 ] ).to.equal( workspacePath );
+			expect( spies.checkout.calledTwice ).to.equal( true );
+			expect( spies.linkDirectories.calledTwice ).to.equal( true );
+			expect( spies.npmInstall.calledTwice ).to.equal( true );
+			expect( spies.installGitHooks.calledTwice ).to.equal( true );
+		} );
+
+		it( 'only checks out repositories if directories are found', () => {
+			const packageJSON = {
+				dependencies: {
+					'ckeditor5-core': 'ckeditor/ckeditor5-core',
+					'ckeditor5-plugin-devtest': 'ckeditor/ckeditor5-plugin-devtest',
+					'non-ckeditor-plugin': 'other/plugin'
+				}
+			};
+
+			spies.getDirectories.restore();
+			spies.getDirectories = sinon.stub( tools, 'getCKE5Directories', () => [ 'ckeditor5-core', 'ckeditor5-plugin-devtest' ] );
+
+			initTask( mainRepositoryPath, packageJSON, options, emptyFn, emptyFn );
+			expect( spies.getDependencies.calledOnce ).to.equal( true );
+			expect( spies.getDependencies.firstCall.args[ 0 ] ).to.equal( packageJSON.dependencies );
+			expect( spies.getDirectories.calledOnce ).to.equal( true );
+			expect( spies.getDirectories.firstCall.args[ 0 ] ).to.equal( path.join( mainRepositoryPath, options.workspaceRoot ) );
+			expect( spies.parseRepositoryUrl.calledTwice ).to.equal( true );
+			expect( spies.cloneRepository.called ).to.equal( false );
+			expect( spies.checkout.calledTwice ).to.equal( true );
+			expect( spies.linkDirectories.calledTwice ).to.equal( true );
+			expect( spies.npmInstall.calledTwice ).to.equal( true );
+			expect( spies.installGitHooks.calledTwice ).to.equal( true );
+		} );
+	} );
+
+	function createSpies() {
```

Comment:
We usually divide test file into two sections, preparations and then TCs. It means that it's semantically OK to put such functions at the end of the first section.